src/Google.Api.Gax/ApiCall.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Google.Protobuf;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Wrapper for matching asynchronous and synchronous API calls to the same underlying method.
+    /// </summary>
+    /// <typeparam name="TRequest">Request type, which must be a protobuf message.</typeparam>
+    /// <typeparam name="TResponse">Response type, which must be a protobuf message.</typeparam>
+    public sealed class ApiCall<TRequest, TResponse>
+        where TRequest : class, IMessage<TRequest>
+        where TResponse : class, IMessage<TResponse>
+    {
+        /// <summary>
+        /// Delegate representing a single asynchronous API call with a request and <see cref="CallSettings"/>.
+        /// </summary>
+        /// <param name="request">Request to make to the service.</param>
+        /// <param name="callSettings">The (optional) settings to apply to this call.</param>
+        /// <returns>A task representing the API call; the result of the completed task will be the API response.</returns>
+        public delegate Task<TResponse> AsyncCall(TRequest request, CallSettings callSettings);
+        /// <summary>
+        /// Delegate representing a single synchronous API call with a request and <see cref="CallSettings"/>.
+        /// </summary>
+        /// <param name="request">Request to make to the service.</param>
+        /// <param name="callSettings">The (optional) settings to apply to this call.</param>
+        /// <returns>The API response.</returns>
+        public delegate TResponse SyncCall(TRequest request, CallSettings callSettings);
+
+        public ApiCall(AsyncCall async, SyncCall sync)
```

Comment:
I'd suggest changing the names to asyncCall and syncCall, partly to avoid using a contextual keyword as an identifier. (It works, but it can screw up syntax highlighting and probably mental reading...)